### ISSUE TITLE
add configure switches for ALSA, JACK and pulseaudio

### ISSUE THREAD

**Issue by domichel** [monocasual/giada#254]
It would be great on linux to have configure options for ALSA, JACK and pulseaudio. The reason is simple: as a musician, linux provide JACK as professional grade sound server and pulseaudio is not even installed into my system. I have to patch Makefile.am in order to remove pulse from the build, and it would be much better if it was enable/disable configure switches for that.

--- giada-0.15.4-src/Makefile.am.orig	2019-03-21 22:43:15.000000000 +0100
+++ giada-0.15.4-src/Makefile.am	2019-07-12 17:32:45.585780900 +0200
@@ -369,11 +369,11 @@
 
 if LINUX
 
-# Add preprocessor flags to enable ALSA, Pulse and JACK in RtAudio.
-cppFlags += -D__LINUX_ALSA__ -D__LINUX_PULSE__ -D__UNIX_JACK__
+# Add preprocessor flags to enable ALSA and JACK in RtAudio.
+cppFlags += -D__LINUX_ALSA__ -D__UNIX_JACK__
 
 ldAdd += -lsndfile -lfltk -lXext -lX11 -lXft -lXpm -lm -ljack -lasound \
-  -lpthread -ldl -lpulse-simple -lpulse -lsamplerate -lrtmidi -ljansson \
+  -lpthread -ldl -lsamplerate -lrtmidi -ljansson \
   -lfreetype -lfontconfig -lXrender -lXfixes -lXcursor -lXinerama
 
 endif

**Comment by monocasual** [comment]
Cool idea. Could you please make a pull request for this?

**Comment by domichel** [comment]
Le Sun, 27 Oct 2019 12:57:47 -0700,
Monocasual Laboratories <notifications@github.com> a écrit :

> Cool idea. Could you please make a pull request for this?
> 

I can try when I get some time. For that, I need to know if, with
rtaudio, which is the real giada dependency on linux, we are free to
enable/disable each one of these 3 audio servers independently of the
other ones?

-- 
If you have a problem and you are not doing anything to fix it, you are
at the heart of the problem.

**Comment by monocasual** [comment]
Sure, actually it's just a matter of defining some preprocessor macros. Specifically: `__LINUX_ALSA__`, `__LINUX_PULSE__` and `__UNIX_JACK__`. 

Currently these macros are hardcoded in the [Makefile.am](https://github.com/monocasual/giada/blob/master/Makefile.am#L381). It's just a matter of mapping new flags such as `--with-alsa`, `--with-pulse` , ... to the macros above.

I can take a look at it as soon as all the issues from 0.16.0-beta1 have been closed.

**Comment by monocasual** [comment]
Postponed to 0.17 as this feature depends on #153 .

**Comment by gvnnz** [comment]
How-to with CMake: https://stackoverflow.com/questions/27570567/is-it-possible-to-set-cmake-variables-with-add-subdirectory/27573961

**Comment by gvnnz** [comment]
Better wait for #315, if PR https://github.com/thestk/rtaudio/pull/251 will get merged. Moving to 0.18.

**Comment by gvnnz** [comment]
Done in cfcd79cdcf4ca805179cad5c577ae399baec1979 .
